app/src/org/odk/collect/android/utilities/NepaliDateUtilities.java

```diff
@@ -0,0 +1,297 @@
+        NEPALI_YEAR_MONTHS.put(2063, new int[] { 0, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2064, new int[] { 0, 31, 31, 32, 32, 31, 30, 30, 29, 30, 29, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2065, new int[] { 0, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31 });
+        NEPALI_YEAR_MONTHS.put(2066, new int[] { 0, 31, 31, 31, 32, 31, 31, 29, 30, 30, 29, 29, 31 });
+        NEPALI_YEAR_MONTHS.put(2067, new int[] { 0, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2068, new int[] { 0, 31, 31, 32, 32, 31, 30, 30, 29, 30, 29, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2069, new int[] { 0, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31 });
+        NEPALI_YEAR_MONTHS.put(2070, new int[] { 0, 31, 31, 31, 32, 31, 31, 29, 30, 30, 29, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2071, new int[] { 0, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2072, new int[] { 0, 31, 32, 31, 32, 31, 30, 30, 29, 30, 29, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2073, new int[] { 0, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 31 });
+        NEPALI_YEAR_MONTHS.put(2074, new int[] { 0, 31, 31, 31, 32, 31, 31, 30, 29, 30, 29, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2075, new int[] { 0, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2076, new int[] { 0, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2077, new int[] { 0, 31, 32, 31, 32, 31, 30, 30, 30, 29, 30, 29, 31 });
+        NEPALI_YEAR_MONTHS.put(2078, new int[] { 0, 31, 31, 31, 32, 31, 31, 30, 29, 30, 29, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2079, new int[] { 0, 31, 31, 32, 31, 31, 31, 30, 29, 30, 29, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2080, new int[] { 0, 31, 32, 31, 32, 31, 30, 30, 30, 29, 29, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2081, new int[] { 0, 31, 31, 32, 32, 31, 30, 30, 30, 29, 30, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2082, new int[] { 0, 30, 32, 31, 32, 31, 30, 30, 30, 29, 30, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2083, new int[] { 0, 31, 31, 32, 31, 31, 30, 30, 30, 29, 30, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2084, new int[] { 0, 31, 31, 32, 31, 31, 30, 30, 30, 29, 30, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2085, new int[] { 0, 31, 32, 31, 32, 30, 31, 30, 30, 29, 30, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2086, new int[] { 0, 30, 32, 31, 32, 31, 30, 30, 30, 29, 30, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2087, new int[] { 0, 31, 31, 32, 31, 31, 31, 30, 30, 29, 30, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2088, new int[] { 0, 30, 31, 32, 32, 30, 31, 30, 30, 29, 30, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2089, new int[] { 0, 30, 32, 31, 32, 31, 30, 30, 30, 29, 30, 30, 30 });
+        NEPALI_YEAR_MONTHS.put(2090, new int[] { 0, 30, 32, 31, 32, 31, 30, 30, 30, 29, 30, 30, 30 });
+    }
+
+    private static final int MAX_YEAR = 2090;
```

Comment:
It's currently thrown in countDaysFromMinDay - if the date to manipulate is out of bounds to start with, an exception is thrown. If, by manipulating the originally in-bounds date, it goes out of bounds, it will just wrap around to the lower end of the available year range.